messaging-client-databus/src/main/java/com/inmobi/messaging/consumer/databus/DatabusConsumer.java

```diff
@@ -255,4 +295,61 @@ protected void createCheckpoint() {
       currentCheckpoint = new CheckpointList(partitionMinList);
     }
   }
+
+  @Override
+  public synchronized void close() {
+    newReaders.clear();
+    if (null != partitionDiscovererAndReaderCreator) {
+      partitionDiscovererAndReaderCreator.cancel();
+    }
+    super.close();
+  }
+
+  protected void startNewReaders() {
+    for (PartitionId id : newReaders.keySet()) {
+      try {
+        PartitionReader reader = newReaders.get(id);
+        reader.start(getReaderNameSuffix());
+        readers.put(id, reader);
+        LOG.info(
+            "started new reader " + getReaderNameSuffix() + " " + reader.toString());
+      } catch (Exception e) {
+        e.printStackTrace();
```

Comment:
Please log exception with a logger and at log level error. 

Is catch required? Can we throw the exceptions out and caller take action. I see Timer task is already catching.